src/dc_strencode.rs

```diff
@@ -25,195 +26,47 @@ use crate::dc_tools::*;
-                        }
-                        end = end.offset(1isize)
-                    }
-                    if *end as libc::c_int != '\u{0}' as i32 {
-                        if mmap_string_append_len(
-                            mmapstr,
-                            end,
-                            cur.wrapping_offset_from(end) as libc::size_t,
-                        )
-                        .is_null()
-                        {
-                            ok_to_continue = false;
-                            continue;
-                        }
-                    }
-                } else if mmap_string_append_len(
-                    mmapstr,
-                    begin,
-                    cur.wrapping_offset_from(begin) as libc::size_t,
-                )
-                .is_null()
-                {
-                    ok_to_continue = false;
-                    continue;
-                }
-                if !(*cur as libc::c_int == ' ' as i32 || *cur as libc::c_int == '\t' as i32) {
-                    continue;
-                }
-                if mmap_string_append_c(mmapstr, *cur).is_null() {
-                    ok_to_continue = false;
-                    continue;
-                }
-                cur = cur.offset(1isize);
-            } else {
-                ret_str = strdup((*mmapstr).str_0);
-                ok_to_continue = false;
-            }
-        }
+pub fn dc_encode_header_words(input: impl AsRef<str>) -> String {
+    let mut result = String::default();
+    for (_, group) in &input.as_ref().chars().group_by(|c| c.is_whitespace()) {
+        let word: String = group.collect();
+        result.push_str(&quote_word(&word.as_bytes()));
     }
 
-    let s = to_string(ret_str);
-    free(ret_str.cast());
-    s
+    result
 }
 
-unsafe fn quote_word(
-    mmapstr: *mut MMAPString,
-    word: *const libc::c_char,
-    size: libc::size_t,
-) -> bool {
-    let mut cur: *const libc::c_char;
-    let mut i = 0;
-    let mut hex: [libc::c_char; 4] = [0; 4];
-    // let mut col: libc::c_int = 0i32;
-    if mmap_string_append(mmapstr, b"=?utf-8?Q?\x00".as_ptr().cast()).is_null() {
-        return false;
-    }
-
-    // col = (*mmapstr).len as libc::c_int;
-    cur = word;
-    while i < size {
-        let mut do_quote_char = false;
-        match *cur as u8 as char {
-            ',' | ':' | '!' | '"' | '#' | '$' | '@' | '[' | '\\' | ']' | '^' | '`' | '{' | '|'
-            | '}' | '~' | '=' | '?' | '_' => do_quote_char = true,
-            _ => {
-                if *cur as u8 >= 128 {
-                    do_quote_char = true;
-                }
-            }
-        }
-        if do_quote_char {
-            print_hex(hex.as_mut_ptr(), cur);
-            if mmap_string_append(mmapstr, hex.as_mut_ptr()).is_null() {
-                return false;
-            }
-        // col += 3i32
+fn quote_word(word: &[u8]) -> String {
+    static ENCODED: phf::Set<u8> = phf_set! {
```

Comment:
phf is gone from our dependencies, please use an alternative. In this case sth like this should do the trick

```rust
fn is_encoded(b: u8) -> bool {
  b == b',' || b == '!' // ... 
}

for byte in word {
  if is_encoded(byte) { .. }
} 
```

Reply:
Okay, will adjust. Mind to link to rationale, why `phf` is removed?